client/sagas/err-saga.js

```diff
@@ -1,11 +1,27 @@
+import combineSagas from '../../common/utils/combine-sagas';
+import { ofType } from '../../common/utils/get-actions-of-type';
 import { makeToast } from '../../common/app/toasts/redux/actions';
+import types from '../../common/app/redux/types';
 
-export default function errorSaga(action$) {
+function errorSaga(action$) {
   return action$
     .filter(({ error }) => !!error)
+    .filter(({ type }) => type !== types.babelError)
     .map(({ error }) => error)
     .doOnNext(error => console.error(error))
     .map(() => makeToast({
       message: 'Something went wrong, please try again later'
     }));
 }
+
+/* this is for catching transpilation errors for the React challenges */
+function babelErrorSaga(action$) {
```

Comment:
The errors were being caught but the issue I saw was the feedback to the user wasn't clear, just "Something went wrong" so I add this saga specifically to catch transpilation errors from the React challenges to indicate to the user that the problem was that their code would not transpile. In the alpha version, each challenges has a specific test to see if the code transpiles or not.